src/main/java/com/google/devtools/build/lib/includescanning/IncludeHintsFunction.java

```diff
@@ -63,20 +65,14 @@ public IncludeHintsFunction(PathFragment hintsFile) {
         return null;
       }
       if (!hintsLookupValue.hasContainingPackage()) {
-        String reasonForNoContainingPackage = hintsLookupValue.getReasonForNoContainingPackage();
-        String message =
-            String.format(
-                "INCLUDE_HINTS file %s was not in a package%s",
-                hintsFile,
-                reasonForNoContainingPackage != null ? ": " + reasonForNoContainingPackage : "");
-        throw new IncludeHintsFunctionException(
-            new EnvironmentalExecException(
-                createFailureDetail(message, Code.INCLUDE_HINTS_FILE_NOT_IN_PACKAGE)));
+        return HintsRules.EMPTY;
       }
       hintsPackageRoot = hintsLookupValue.getContainingPackageRoot();
-      env.getValueOrThrow(FileValue.key(RootedPath.toRootedPath(hintsPackageRoot, hintsFile)),
-          IOException.class);
-    } catch (IOException | BuildFileNotFoundException e) {
+      env.getValueOrThrow(
```

Comment:
What is the plan for tools/cpp/INCLUDE_HINTS in Bazel? The file is at kind of a strange place for Bazel because it's the only place where Bazel reads a file from a specific location in the main repository.

Can we get away with never reading it so that the Bazel code path is a bit simpler?

Reply:
I intended that `INCLUDE_HINTS` be functional with Bazel because it may be useful for third party code with computed includes. That said, even though you can provide a `INCLUDE_HINTS` file under this CL, it's fairly useless, since, for example, it doesn't accept hints outside of a hardcoded `third_party/` directory. I thought that could be fixed in the future if desired.

I could bypass the include hints code in Bazel only. However, working on workspaces that don't have an `INCLUDE_HINTS` file isn't a wild extension in functionality or code complexity for `IncludeHintsFunction`; `git` says `IncludeHintsFunction.java` has 12 deletions and 10 insertions in this CL.

On a more technical note, I don't know what the preferred method to make a product-based branch for this code would be. A new semantics method? `BlazeRuntime.getProductName().equals("Bazel")`?

Reply:
I dont mind too much; since this is experimental functionality, ugliness like reading a random file with a random name from the main repository is not too bad, but I'd still prefer to code to be clean if we can help.

Looking at the code, I think the best alternative would be to set `IncludeScanningModule.INCLUDE_HINTS_FILENAME` to null (we already override it within Google so it would be a no-op for our monorepo) and change `IncludeScanningModule.getSkyFunctions()` and `IncludeHintsFunction` so that if the value is null, they simply return the empty include hints.  Looks like this is doable with a minimal amount of effort.

Reply:
I've now implemented your suggestion to `IncludeScanningModule` and reverted the changes to `IncludeHintsFunction`.

Reply:
Thanks! One fewer hackj.